client-index/src/auto_sync_core.rs

```diff
@@ -217,7 +217,16 @@ where
     pub fn get_queue(&self) -> Sender<OwnedMessage> {
         self.my_sender.clone()
     }
-
+    fn update_wallet(&mut self, info: &WalletInfo) -> Result<()> {
+        for i in 0..self.wallets.len() {
```

Comment:
Instead of checking duplicates in a `Vec`. Can we have a `HashSet` or `BTreeSet` of `WalletInfo`?

Reply:
order is important, so need to use vec or list

Reply:
Why is order important?

Reply:
yes , BTreeMap will be good